tuner/tuner/candidate_gen.py

```diff
@@ -197,6 +244,22 @@ def generate_configs_and_td_specs(
         tune_logger.debug(f"Solution #{i+1}: {config}")
         td_spec_module = dispatch_tuner.get_td_spec(input_module, config)
         assert td_spec_module, "Failed to generate transform dialect spec"
+
+        td_specs_to_link, new_warned_matchers = determine_td_specs_to_link(
+            [starter_td_spec, td_spec_module]
+            if starter_td_spec is not None
+            else [td_spec_module],
```

Comment:
Make this list a local variable and append to it instead of having this branching logic